app/src/main/java/com/andrew/tasky/core/presentation/main/MainActivity.kt

```diff
@@ -24,11 +25,14 @@ class MainActivity : AppCompatActivity() {
 
         installSplashScreen().apply {
             setKeepOnScreenCondition {
-                if (viewModel.isUserLoggedIn.value) {
-                    navController.navigate(R.id.action_global_agendaFragment)
-                }
                 viewModel.isLoading.value
             }
         }
+
+        collectLatestLifecycleFlow(viewModel.isUserLoggedIn) { isLoggedIn ->
+            if (!isLoggedIn) {
+                navController.navigate(R.id.action_global_loginFragment)
```

Comment:
Here you might want to also clear the backstack, since otherwise the user could press back on the login screen and get back to the agenda screen if this gets triggered from there